### PR TITLE
Notes about 10.15 changes

### LINKED ISSUE #633: Update documentation for developing plugins supporting macOS 10.15
Plugins containing native frameworks or binaries must be notarised in order to work on macOS 10.15 Catalina. Provide a guide for plugin developers how to go about it using `skpm` or notarise manually on the command-line.

### PR DESCRIPTION
**Pull request by bomberstudios**
Fixes #633

Credits to @abynim for some of the information

### CODE REVIEW

File: docs/pages/plugins-changes-in-catalina.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+2. Drop your framework into your .sketchplugin bundle and zip the entire bundle.
+3. Code-sign the zip using this command in the terminal. The identifier must match the bundle identifier of your plugin framework (not necessarily your plugin identifier in the manifest).
+
+    ```bash
+    codesign -f -s "Developer ID Application: John Doe" --timestamp --identifier "com.organization.PluginName" /Users/johndoe/Desktop/PluginName.zip
+    ```
+
+4. Send the zip for notarization:
+    ```bash
+    xcrun altool --notarize-app -f /Users/johndoe/Desktop/PluginName.zip --primary-bundle-id "com.organization.PluginName" -u "yourAppleIDEmail@gmail.com" -p "YourAppleIDPassword"
```
Comment (mathieudutour) [inline docs/pages/plugins-changes-in-catalina.md]: `YourAppleIDPassword` needs to be an app-specific password (same as https://github.com/BohemianCoding/SketchAPI/pull/635/files#diff-cc2fc8a2c3786dc437de276d7ec2ec5cR44), so it's not exactly the apple ID password

Comment (bomberstudios) [reply docs/pages/plugins-changes-in-catalina.md]: 🙌

File: docs/pages/plugins-changes-in-catalina.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+4. Send the zip for notarization:
+    ```bash
+    xcrun altool --notarize-app -f /Users/johndoe/Desktop/PluginName.zip --primary-bundle-id "com.organization.PluginName" -u "yourAppleIDEmail@gmail.com" -p "wzac-jvbe-nhqu-wjao"
+    ```
+
+5. Apple will then send you an email in about 3-4 minutes that your bundle is notarized. You can now release the plugin.
+6. If you make any changes to your plugin framework you’ll need to notarize again, so do all this as the last step before release.
+
+## Further reading
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]: We use _Related resources_ in [other places](https://developer.sketch.com/plugins/cocoascript) and _Next steps_, for consistency I suggest to use the one that works better.

```suggestion
## Related resources
```

File: docs/pages/plugins-changes-in-catalina.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+- [Notarizing your app before distribution](https://developer.apple.com/documentation/xcode/notarizing_your_app_before_distribution), at Apple's Developer Site.
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]:
```suggestion
- [Notarizing Your App Before Distribution](https://developer.apple.com/documentation/xcode/notarizing_your_app_before_distribution) Apple Documentation
```

File: docs/pages/plugins-changes-in-catalina.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+(Credits to Aby Nimbalkar for the information)
+
+Requirements for notarizing plugins using Xcode and the command line:
+
+- [Two-factor authentication](https://support.apple.com/en-us/HT204915) needs to be enabled for your account.
+- An [app-specific password](https://support.apple.com/en-us/HT204397)to use in Terminal for the notarization. It will have a format like this: `wzac-jvbe-nhqu-wjao`
+
+1. Make sure your plugin framework is signed as a Developer ID Application in Xcode:
+
+    ![](/images/developer/dev-id.png)
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]: To make sure it looks nice on retina screens please use `img` tag with width specified and height on `auto` see, other pages.

File: docs/pages/plugins-changes-in-catalina.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+```yaml
+notarization:
+  command: './custom-script.sh'
+```
+
+Skpm will execute `./custom-script.sh ./path/to/plugin.zip` and expect the same zip to be notarized at the end.
+
+
+## Notarizing plugins using Xcode
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]:
```suggestion
## 2. Use Xcode
```

File: docs/pages/plugins-changes-in-catalina.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    ```bash
+    codesign -f -s "Developer ID Application: John Doe" --timestamp --identifier "com.organization.PluginName" /Users/johndoe/Desktop/PluginName.zip
+    ```
+
+4. Send the zip for notarization:
+    ```bash
+    xcrun altool --notarize-app -f /Users/johndoe/Desktop/PluginName.zip --primary-bundle-id "com.organization.PluginName" -u "yourAppleIDEmail@gmail.com" -p "wzac-jvbe-nhqu-wjao"
+    ```
+
+5. Apple will then send you an email in about 3-4 minutes that your bundle is notarized. You can now release the plugin.
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]:
```suggestion
1. Wait for a notarization email confirmation from Apple before releasing the plugin.
```

File: docs/pages/plugins-changes-in-catalina.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+6. If you make any changes to your plugin framework you’ll need to notarize again, so do all this as the last step before release.
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]:
```suggestion
> **Note:** If you make any changes to your plugin framework you’ll need to notarize again.
```

File: docs/pages/plugins-changes-in-catalina.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+
+## Notarizing plugins using Xcode
+
+(Credits to Aby Nimbalkar for the information)
+
+Requirements for notarizing plugins using Xcode and the command line:
+
+- [Two-factor authentication](https://support.apple.com/en-us/HT204915) needs to be enabled for your account.
+- An [app-specific password](https://support.apple.com/en-us/HT204397)to use in Terminal for the notarization. It will have a format like this: `wzac-jvbe-nhqu-wjao`
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]: Aren't we mentioning this earlier already and can be seen as general prerequisites?

File: docs/pages/plugins-changes-in-catalina.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+notarization:
+  command: './custom-script.sh'
+```
+
+Skpm will execute `./custom-script.sh ./path/to/plugin.zip` and expect the same zip to be notarized at the end.
+
+
+## Notarizing plugins using Xcode
+
+(Credits to Aby Nimbalkar for the information)
```
Comment (christianklotz) [inline docs/pages/plugins-changes-in-catalina.md]: Mention acknowledgement in PR and release notes instead

File: docs/pages/plugins-changes-in-catalina.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+## Notarizing plugins using `skpm`
+
+(From <https://github.com/skpm/skpm/blob/master/docs/notarization.md>)
+
+`skpm` makes it easy to notarize your plugin when you publish it.
+
+### How to
+
+If you do not have a native framework, skip to 2.
```
[the remark this answers is not in the document]
Comment (bomberstudios) [reply docs/pages/plugins-changes-in-catalina.md]: Maybe this is necessary if you are bundling any third party binaries that are not native frameworks? I assume those would need to be notarised as well 🤔 Just thinking aloud here, but I just realised that _if what I just said is true_ then we need to check our own sample plugins to see if we have any bundled binary (I think we don't, but it doesn't hurt to check).

I'll make a quick test plugin with bundled binaries to verify this.

**Comment by bomberstudios** [comment]
Changed the page name to better reflect the contents, but the final name is pending my investigation about plugins that use bundled binaries that are not frameworks (in which case, the title and filename will change to just 'Notarizing Plugins')

File: docs/pages/plugins-changes-in-catalina.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+     authority: 'Developer ID Application: TEAM'
+     username: 'AC_USERNAME'
+     password: '@keychain:AC_PASSWORD'
+   ```
+
+Replace `TEAM` with the name of your team on App Store Connect.
+
+Replace the `AC_USERNAME` with your App Store Connect username (usually an email address). Because App Store Connect now requires two-factor authentication (2FA) on all accounts, you must create an app-specific password, as described in [Using app-specific passwords](https://support.apple.com/en-us/HT204397).
+
+To avoid including your password as cleartext in the file, you can provide a reference to a keychain item. This assumes the keychain holds a keychain item named `AC_PASSWORD` with an account value matching the username `AC_USERNAME`. Note that skpm can’t access your iCloud keychain for security reasons, so the item must be in your login keychain. You can add a new keychain item using the Keychain Access app, or from the command line using the security utility:
```
[the remark this answers is not in the document]
Comment (bomberstudios) [reply docs/pages/plugins-changes-in-catalina.md]: Not really: there is a way to specify the password without using the keychain, which is to have it directly on the `.skpmrc` file. I'll reword this section to make everything clearer.

File: docs/pages/plugins-changes-in-catalina.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (bomberstudios) [reply docs/pages/plugins-changes-in-catalina.md]: My worst fears have turned out to be true… Catalina shows this when running a plugin with an embedded binary:

Screenshot 2019-10-07 at 18 37 19

[the remark this answers is not in the document]
Comment (bomberstudios) [reply docs/pages/plugins-changes-in-catalina.md]: Just checked with Mathieu, and the notarization feature in `skpm` should take care of this, so I'm tweaking the page title and URL

File: docs/pages/plugins-notarize-plugins.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+            3. Set the password using your app specific password.  
+
+> **Quick tip:** To use your own notarization mechanism and integrate with `skpm`,
+provide the command in the `.skpmrc` notarization settings:
+>
+> ```yaml
+> notarization:
+>   command: './custom-script.sh'
+> ```
+> `skpm` will bundle and zip your plugin, run your command automatically, and set the path to the *zipped* plugin bundle as a parameter, e.g. `./notarize-plugin.sh path/to/plugin.zip`.
```
Comment (christianklotz) [inline docs/pages/plugins-notarize-plugins.md]:
```suggestion
> `skpm` will bundle your plugin, create a ZIP archive of it, and run your command automatically, passing the path to the archive as a parameter, e.g. `./notarize-plugin.sh path/to/select-shapes.sketchplugin-1.0.zip`.
```

**Comment by bomberstudios** [comment]
I am _really_ tempted to remove the whole "Xcode Signing Settings" section. I assume if someone is using a native framework on their plugin, they'll already know how to sign it (otherwise it wouldn't work anywhere!), and it just adds noise to the documentation IMHO

**Pull request merged**